### ISSUE TITLE
Satellite images(tif)

### ISSUE THREAD

**Issue by hakanerten09** [milesial/Pytorch-UNet#249]
Hi. Can i use tif satellite images for dataset. if answer is yes, please help me. How can i use and prepare my dataset?

**Comment by TimandXiyu** [comment]
Not sure if you are working with a super large image or something else.
**But pillow should be able to load the image anyway.**

And this repo is using pillow for Dataloader. 
So, what do you mean by saying preparing the dataset? 
Do you mean you want to randomly cut the big image into small data points?

**Comment by hakanerten09** [comment]
hi @TimandXiyu ,
Thanks for quick reply. 
I had super large image but finally i cut the big image to 4096x2048 pieces. i also have the labelled images. But i haven't annotations. Only images and labelled images. Is it available for your code?

**Comment by TimandXiyu** [comment]
I think that 4096x2048 is still too large for 24GB TITAN or 3090 cards.
You better cut it down to something like 512*512 randomly. (It means random position and random angle, there should be plenty of git repos that coud do that)

What is your segmentation task about? Road? Buildings? or Multiple Objection Segmentation?
What is your label looks like? Can you briefly summarize that? Is it binary or what?

I also works with satellite images, but mainly for raod extraction. So I only need binary masks.
BTW, if road extraction is your case, I would recommend Dice loss or IoU loss instead of standard BCEwithlogits.

**Comment by hakanerten09** [comment]
Ok. I can cut it down smaller.
I am working with sar images. SAR images resolution is bad for segmentation. My classes are :  ('forest', 'urban', 'agricultural_areas', 'others', 'water'). I labelled my sar images myself with some programs. I only have images and labelled images with the same resolution and coordinates.

**Comment by TimandXiyu** [comment]
So, since you do have the masks, and the image. I think that should be enough for typical deep learning algorithms?

Just randomly cut the image and mask simultaneously. (doing the same operations to both the image and mask, then you have a data and label pair)

I am not sure if I am getting your question here correctly.

**Comment by hakanerten09** [comment]
Thanks a lot,
It is enough for now. I will try and share my results.
